SAML Integrations (#807)
clerk/clerk-docs: docs/authentication/saml/google.mdx

2. In the navigation sidebar, under **Apps**, select **Web and mobile apps**.
3. Select the **Add app** button.
4. From the dropdown, select **Add custom SAML app**.
5. In the **App details** section, an **App name** is required. You can also add an **App description** and upload an **App icon** if you wish.
6. Select the **Continue** button.

### Configure your identity provider

There are two options for configuring your IdP:

- [**Metadata configuration**](#metadata-configuration) - This is where you can input the URL to your IdP's metadata file. This is the recommended way to configure your IdP.
- [**Custom configuration**](#custom-configuration) - This is where you can manually input the configuration settings for your IdP.

#### Metadata configuration

1. In the Google Admin Console, under **Option 1: Download IdP Metadata**, select the **Download Metadata** button.
2. In the Clerk Dashboard, find the **Identity Provider configuration** section.
3. Under the **Metadata configuration** option, select the **Upload** button.
4. Upload the metadata file you downloaded from Google.
5. Select the **Save changes** button to complete the setup.

#### Custom configuration

> If you choose to manually input the configuration settings for your IdP, you will need to add these three fields to your Clerk settings:

Reviewer: ```suggestion
If you choose to manually input the configuration settings for your IdP, you must add these three fields to your Clerk settings:
```

- **SSO URL** - This is the unique identifier of your IdP application.
- **Entity ID** - This is your IdP's URL that Clerk will redirect your users to so that they can authenticate.
- **Certificate** - This is the certificate needed for Clerk to securely connect to your IdP.

1. In the Google Admin Console, under **Option 2**, copy the **SSO URL**, **Entity ID**, and download the **Certificate**.
2. In the Clerk Dashboard, find the **Identity Provider configuration** section.
3. Select the **Custom configuration** option.
4. Fill in the **SSO URL**, **Entity ID**, and upload the **Certificate**.
5. Select the **Save** button in the Clerk Dashboard.
6. Select the **Continue** button in the Google Admin Console.

### Configure your service provider

> To configure your service provider (Clerk), you will need to add these three fields to your IdP's application:

Reviewer: ```suggestion
To configure your service provider (Clerk), you must add these three fields to your IdP's application:
```

Anywhere we say "will need to", we should say "must", as it's less wordy

> ---

Reviewer 2: We should Google Workspace as the IdP name in this page

> title: Integrate Google SAML with Clerk
> description: Learn how to integrate .

Reviewer 2: ```suggestion
title: Google Workspace
description: Learn how to integrate Google Workspace via SAML SSO.
```

---

> # Integrate Google SAML with Clerk

Reviewer 2: ```suggestion
# Google Workspace
```

<TutorialHero
  beforeYouStart={[
    {
      title: "Set up a Clerk application",
      link: "/docs/quickstarts/setup-clerk",
    },
    {
      title: "Set up a local project",
      link: "/docs/quickstarts/overview",
    },
  ]}
>

> - Use Google SAML to authenticate users in your Clerk application.

Reviewer 2: ```suggestion
- Use Google Workspace to enable single sign-on (SSO) via SAML for your Clerk application.
```

</TutorialHero>

> Integrating Google SAML with Clerk process involves setting up a SAML application in Google Cloud and configuring it to work with Clerk. This tutorial will guide you through the process of setting up Google SAML and integrating it with Clerk.

Reviewer 2: ```suggestion
Integrating Google Workspace with Clerk allows you to use it as an Identity Provider for your Clerk application to authenticate your users and manage their access to your application via SAML SSO. This tutorial will guide you through the process of setting up Google SAML and integrating it with Clerk.
```

## Tutorial

<Callout type="info">
  You must have a local project with Clerk set up already to follow this tutorial. See [the quickstart docs](/docs/quickstarts/overview) to get started.
</Callout>

<Steps>

> ### Setup Clerk SAML Connection

Reviewer 2: ```suggestion
### Setup Clerk Enterprise Connection
```

<Images 
  width={3024}
  height={1652}
  src="/docs/images/authentication/saml/.avif"
  alt=""
/>

###  Create a new Google Enterprise Application

<Images 
  width={3024}
  height={1652}
  src="/docs/images/authentication/saml/.avif"
  alt=""
/>

#### Follow the Google SAML Setup Wizard

<Images 
width={3024} 
height={1652} 
src="/docs/images/authentication/saml/.avif" 
/>

<Images 
width={3024} 
height={1652} 
src="/docs/images/authentication/saml/.avif" 
/>

<Images 
width={3024} 
height={1652} 
src="/docs/images/authentication/saml/.avif" 
/>

SAML Configuration in Google 

<Images
  width={3024}
  height={1652}
  src="/docs/images/authentication/saml/.avif"
  alt=""
/>

#### Clerk Reference for SAML Settings

<Images
  width={3024}
  height={1652}
  src="/docs/images/authentication/saml/.avif"
  alt=""
/>

### Attribute Mapping in Google to Clerk 

You will want to add the following fields to the attribute mapping in Google to Clerk.
<Images 
width={3024} 
height={1652} 
src="/docs/images/authentication/saml/.avif"
alt="Attribute Mapping in Google to Clerk" 
/>

<Images 
  width={3024}
  height={1652}
  src="/docs/images/authentication/saml/.avif"
  alt=""
/>

> ### Google PEM Certificate Download & Upload to Clerk

Reviewer 2: We should guide users to download the connection metadata file and use that to configure it in the Clerk Dashboard instead of the manual process

Author: Mmmm I see, yeah during your Demo I noticed the one from Sokratis was a bit outdated
